java/http/vertx/java-http-vertx-producer/src/main/java/ProducerApp.java

```diff
@@ -3,64 +3,51 @@
  */
 
+import java.util.HashMap;
 import java.util.Map;
 import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.TimeUnit;
 
-import io.jaegertracing.Configuration;
-import io.opentracing.Tracer;
-import io.opentracing.util.GlobalTracer;
-import org.apache.logging.log4j.Logger;
-import org.apache.logging.log4j.LogManager;
-
-import io.vertx.config.ConfigRetriever;
-import io.vertx.config.ConfigRetrieverOptions;
-import io.vertx.config.ConfigStoreOptions;
 import io.vertx.core.Vertx;
-import io.vertx.core.json.JsonObject;
+import io.vertx.core.VertxOptions;
+import io.vertx.tracing.opentelemetry.OpenTelemetryOptions;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import io.strimzi.common.TracingSystem;
 
 public final class ProducerApp {
-    
-    private static final Logger log = LogManager.getLogger(ProducerApp.class);
 
-    private static String deploymentId;
+    private static final Logger log = LogManager.getLogger(ProducerApp.class);
 
     public static void main(String[] args) throws Exception {
-        Vertx vertx = Vertx.vertx();
-
         CountDownLatch messagesSentLatch = new CountDownLatch(1);
         CountDownLatch exitLatch = new CountDownLatch(1);
 
-        ConfigStoreOptions envStore = new ConfigStoreOptions()
-                .setType("env")
-                .setConfig(new JsonObject().put("raw-data", true));
-
-        ConfigRetrieverOptions options = new ConfigRetrieverOptions()
-                .addStore(envStore);
-
-        ConfigRetriever retriever = ConfigRetriever.create(vertx, options);
-
-        retriever.getConfig(ar -> {
-            Map<String, Object> envConfig = ar.result().getMap();
-            HttpKafkaProducerConfig httpKafkaConsumerConfig = HttpKafkaProducerConfig.fromMap(envConfig);
-
-            HttpKafkaProducer httpKafkaProducer = new HttpKafkaProducer(httpKafkaConsumerConfig, messagesSentLatch);
-
-            vertx.deployVerticle(httpKafkaProducer, done -> {
-                if (done.succeeded()) {
-                    deploymentId = done.result();
-
-                    if (envConfig.get("JAEGER_SERVICE_NAME") != null) {
-                        Tracer tracer = Configuration.fromEnv().getTracer();
-                        GlobalTracer.registerIfAbsent(tracer);
-                    }
-                    
-                    log.info("HTTP Kafka producer started successfully");
-                } else {
-                    log.error("Failed to deploy HTTP Kafka producer", done.cause());
-                    System.exit(1);
-                }
-            });
+        Map<String, Object> objectMap = new HashMap<>();
+        for (Map.Entry<String, String> entry : System.getenv().entrySet()) {
+            objectMap.put(entry.getKey(), entry.getValue());
+        }
+        HttpKafkaProducerConfig httpKafkaProducerConfig = HttpKafkaProducerConfig.fromMap(objectMap);
```

Comment:
as for the consumer, let's follow the `fromEnv` approach as the other Java HTTP examples.

Reply:
Done